packages/react/runtime/src/debug/profile.ts

```diff
@@ -1,52 +1,174 @@
 // Copyright 2024 The Lynx Authors. All rights reserved.
 // Licensed under the Apache License Version 2.0 that can be found in the
 // LICENSE file in the root directory of this source tree.
-import { options } from 'preact';
-import type { Component, ComponentClass, VNode } from 'preact';
+import { Component, options } from 'preact';
+import type { ComponentClass, VNode } from 'preact';
 
-import { COMPONENT, DIFF, DIFFED, RENDER } from '../renderToOpcodes/constants.js';
-import { getDisplayName } from '../utils.js';
+import type { TraceOption } from '@lynx-js/types';
+
+import { globalPatchOptions } from '../lifecycle/patch/commit.js';
+import { __globalSnapshotPatch } from '../lifecycle/patch/snapshotPatch.js';
+import { COMMIT, COMPONENT, DIFF, DIFF2, DIFFED, DIRTY, NEXT_STATE, RENDER } from '../renderToOpcodes/constants.js';
+import { getDisplayName, hook } from '../utils.js';
 
 export function initProfileHook(): void {
-  const oldDiff = options[DIFF];
-  options[DIFF] = function(vnode: VNode) {
-    // This __PROFILE__ is used for DCE testing
-    if (__PROFILE__ && typeof vnode.type === 'function') {
-      // We only add profiling trace for Component
-      console.profile(`diff::${getDisplayName(vnode.type as ComponentClass)}`);
+  // early-exit if required profiling APIs are unavailable
+  let p;
+  /* v8 ignore start */
+  if (
+    !(p = lynx.performance)
+    || typeof p.profileStart !== 'function'
+    || typeof p.profileEnd !== 'function'
+    || typeof p.profileMark !== 'function'
+    || typeof p.profileFlowId !== 'function'
+  ) {
+    return;
+  }
+  /* v8 ignore stop */
+
+  const profileStart = p.profileStart.bind(p);
+  const profileEnd = p.profileEnd.bind(p);
+  const profileMark = p.profileMark.bind(p);
+  const profileFlowId = p.profileFlowId.bind(p);
+
+  // for each setState call, we will add a profiling trace and
+  // attach a flowId to the component instance.
+  // This allows us to trace the flow of its diffing, committing and patching.
+  {
+    const sFlowID = Symbol.for('FLOW_ID');
```

Comment:
If we are not using `sFlowID` elsewhere, can we use `Symbol('FLOW_ID')` instead?
```suggestion
    const sFlowID = Symbol('FLOW_ID');
```